### PR TITLE
Ruby Deprecation doc

### PR DESCRIPTION
**Pull request by kairu-ms**

### CODE REVIEW

File: docs/README.md
Comment (nickzhums) [inline docs/README.md:3]:
```suggestion
As of (2022/01), Azure SDK for Ruby - Resource Management Libraries are officially deprecated. The libraries will be in maintenance mode for a period of 3 years and security / bug fixes will still be applied. However, there will be no further releases based on feature requests. 

We recommend you take one of the following approaches
- Use Azure REST API directly in Ruby (instructions are provided in this doc)
- Use one of the next-generation SDKs that provides more idiomatic, consistent, approachable, dependable, and diagnosable experience. You can find more info on the repo of each individual language
  - [Java](https://github.com/Azure/azure-sdk-for-java/#management-new-releases)
  - [.NET](https://github.com/Azure/azure-sdk-for-net/#management-new-releases)
  - [Python](https://github.com/Azure/azure-sdk-for-python/#management-new-releases)
```

File: docs/README.md
Comment (nickzhums) [inline docs/README.md:9]:
```suggestion
You can also find information about the latest SDK releases as well as API references on the [official Azure SDK web-site](https://azure.github.io/azure-sdk/)
```

File: docs/README.md
Comment (nickzhums) [inline docs/README.md:11]:
```suggestion
# Using Azure REST API in Ruby
```

File: docs/README.md
Comment (nickzhums) [inline docs/README.md:14]:
```suggestion
## Reference
[Azure REST API Reference](https://docs.microsoft.com/en-us/rest/api/azure/)
```

File: docs/README.md
Comment (nickzhums) [inline docs/README.md:18]: so we are still relying on ms_rest_azure to use the REST API ? if this is the case, we probably should mention it at the beginning as well

Comment (nickzhums) [inline docs/README.md:18]: The first step to calling the REST API is authentication. For people who are unfamiliar with Azure, this maybe one of the challenging concepts. 

To authenticate, we will need to set up a Service Principal. For reference on how to set up a Service Principal from command line, please see [Authenticating a service principal with Azure Resource Manager](http://aka.ms/cli-service-principal) or [Unattended Authentication](http://aka.ms/auth-unattended).

For a more robust explanation of authentication in Azure, please follow  [Developer’s guide to auth with Azure Resource Manager API](http://aka.ms/arm-auth-dev-guide).

After creating the Service Principal, you should have the following key identifiers
- Client ID (GUID)
- Client Secret (String)
- Tenant ID (GUID) or domain name (String)

File: docs/README.md
Comment (nickzhums) [inline docs/README.md:34]: consider moving "The ms_rest_azure gem is a Azure rest client library used by Azure SDK for Ruby. It relies on faraday and concurrent-ruby. " into the Install section

File: docs/README.md
Comment (nickzhums) [inline docs/README.md:71]: In addition to environment variables, you can also pass those parameters in an Option hash when you initialize the profile clients

File: docs/README.md
Comment (nickzhums) [inline docs/README.md:86]: maybe provide a little bit background info on how token works in Azure

File: docs/README.md
Comment (nickzhums) [inline docs/README.md:36]: what's the current status of ms_rest_azure, is it also being deprecated?

Comment (nickzhums) [reply docs/README.md:36]: or is someone maintaining it?

File: docs/README.md
Comment (nickzhums) [inline docs/README.md:149]: is async request the standard pattern in Ruby? if so, we should provide an async request code sample

File: docs/README.md
Comment (nickzhums) [inline docs/README.md:151]: Code Sample

File: docs/README.md
Comment (nickzhums) [inline docs/README.md:253]: we should provide one more sample in addition to checking storage availability. A classical sample is creating a virtual machine as it contains creating a resource group, and network

File: docs/README.md
Comment (nickzhums) [inline docs/README.md:254]: refine this section to be more like https://github.com/Azure/azure-sdk-for-java/#need-help

File: docs/README.md
Comment (yonzhan) [reply docs/README.md:27]: Thanks Nick. It seemed that above link about Developer’s guide to auth with Azure Resource Manager API could not be opened.

File: docs/README.md
Comment (kairu-ms) [reply docs/README.md:158]: All the samples will be asynchronized.

File: docs/README.md
Comment (kairu-ms) [reply docs/README.md:262]: I will add additional markdown files for that.

File: docs/README.md
Comment (kairu-ms) [reply docs/README.md:45]: ms_rest_azure is the runtime part of Ruby SDK. I think we can keep maintaining this library. It's stable, light weighted and easy to use.

File: docs/README.md
Comment (nickzhums) [inline docs/README.md:14]: we can remove those redundant repo links

File: docs/README.md
Comment (nickzhums) [inline docs/README.md:20]:
```suggestion
## API Reference
```

File: docs/README.md
Comment (nickzhums) [inline docs/README.md:21]: I suggest moving API reference section to between authentication and making the REST call

File: docs/README.md
Comment (nickzhums) [inline docs/README.md:41]:
```suggestion
## Make your REST request
```

File: docs/README.md
Comment (nickzhums) [inline docs/README.md:116]: are token providers mandatory in our scenario?  if so, we should make it more clear like Step 1 -> Step 2 -> Step 3

File: docs/README.md
Comment (nickzhums) [inline docs/README.md:280]:
```suggestion
## Additional Code Samples
```

**Comment by nickzhums** [comment]
@kairu-ms  no further comments from me, in the meantime,let's hold off on merging this PR until i get a confirmation on final approval deprecation, thanks!

File: docs/README.md
Comment (serprex) [inline docs/README.md:110]:
```suggestion
### Step 1 [Acquire an access token](https://docs.microsoft.com/en-us/rest/api/azure/#acquire-an-access-token)
```

**Review by blaquezzz — APPROVED** [review]

File: docs/README.md
Comment (ItalyPaleAle) [inline docs/README.md:7]: No we can't say that. If customers are using Ruby, we can't tell them to switch language

File: docs/README.md
Comment (kairu-ms) [inline docs/README.md:10]:
```suggestion
We recommend you use Azure REST API directly in Ruby (instructions are provided in this doc)
```

File: docs/README.md
Comment (kairu-ms) [reply docs/README.md:7]: > No we can't say that. If customers are using Ruby, we can't tell them to switch language

OK. I will remove this paragraph.

**Comment by nickzhums** [comment]
@kairu-ms @ItalyPaleAle  i made a few changes in the branch to update readme, please review

**Comment by kairu-ms** [comment]
> @kairu-ms @ItalyPaleAle i made a few changes in the branch to update readme, please review

It looks good to me.

File: README.md
Comment (ItalyPaleAle) [inline README.md:11]: "Here is the complete list" -> "Below is the complete list"

Comment (ItalyPaleAle) [inline README.md:11]: "After 2021" -> "After that date"

Comment (ItalyPaleAle) [inline README.md:11]: "12/31/2021" -> "2021-12-31" or "December 31, 2021" because outside of the US "12/31" while possibly un-ambiguous would still look odd :)

File: README.md
Comment (ItalyPaleAle) [inline README.md:21]: ")." -> "),"

File: docs/README.md
Comment (ItalyPaleAle) [inline docs/README.md:5]: "In the future, we recommend that you interface with the Azure REST APIs directly from Ruby; instructions are provided in this doc."

**Review by ItalyPaleAle — APPROVED** [review]
Looks good, just some minor suggestions

**Comment by nickzhums** [comment]
Thanks @ItalyPaleAle . I updated the doc per your suggestions

**Pull request closed without merging**
